### PR TITLE
Potentially fixes a cowboy boot snake harddel

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

image
Yeeeeeah

Cowboy boots didn't have any proc for dumping occupants upon being destroyed so it caused a harddel. I think.

## Changelog

:cl:
fix: Potentially fixed a cowboy boot snake harddel
/:cl:

### CODE REVIEW

**Review by Mothblocks — CHANGES_REQUESTED** [review]
image

This PR will not be merged until you add a regression test to make sure this doesn't happen again. You should have done this before writing the code for the fix.

This is, luckily, easy. Add a variable that determines whether or not the random chance rolls consistently.

Then, in the create_and_destroy test, add a new section at the bottom like so, because if we are doing our jobs then we will eventually fill a LOT of these:

```dm
// These are type paths that exist for the purpose of reproducing inconsistent behavior

/// Cowboy boots that consistently spawn a snake in the boot
/obj/item/clothing/shoes/cowboy/consistent
    consistent_snake_in_boot = TRUE
```

**Comment by Mothblocks** [comment]
> Cowboy boots didn't have any proc for dumping occupants upon being destroyed so it caused a harddel. I think.

A great reason for everything I just said--you literally don't even know if this works

File: code/modules/clothing/shoes/cowboy.dm [hunk 1]
```diff
@@ -15,6 +15,10 @@
 		var/mob/living/simple_animal/hostile/retaliate/snake/bootsnake = new/mob/living/simple_animal/hostile/retaliate/snake(src)
 		occupants += bootsnake
 
+/obj/item/clothing/shoes/cowboy/Destroy()
+	for(var/mob/living/occupant in occupants)
+		occupant.forceMove(loc)
```
Comment (LemonInTheDark) [inline code/modules/clothing/shoes/cowboy.dm:20]: Assertion: The snake is already inside our contents, and thus will already be qdeleted on destroy.
So the issue here is the occupants list failing to update. 
You should use Exited to remove the snake from the list when it leaves, that way there's no potential gottchas for deleting the snake or something

**Comment by Mothblocks** [comment]
OP said they can't make the time to fix this so I'm reopening the branch I was working on

**Pull request closed without merging**